models/dpr_test/dpr_test.c

```diff
@@ -0,0 +1,238 @@
+  if (IS_ERR(fe_dpr_test_devp->regs)) {
+    ret_val = PTR_ERR(fe_dpr_test_devp->regs);
+    goto bad_exit_return;
+  }
+  platform_set_drvdata(pdev, (void *)fe_dpr_test_devp);
+  fe_dpr_test_devp->name = devm_kzalloc(&pdev->dev, 50, GFP_KERNEL);
+  if (fe_dpr_test_devp->name == NULL)
+    goto bad_mem_alloc;
+  strcpy(fe_dpr_test_devp->name, (char *)pdev->name);
+  pr_info("%s\n", (char *)pdev->name);
+  status = alloc_chrdev_region(&dev_num, 0, 1, "fe_dpr_test_");
+  if (status != 0)
+    goto bad_alloc_chrdev_region;
+  sprintf(deviceMinor, "%d", MAJOR(dev_num));
+  strcat(device_name, deviceMinor);
+  pr_info("%s\n", device_name);
+  cl = class_create(THIS_MODULE, device_name);
+  if (cl == NULL)
+    goto bad_class_create;
+  cdev_init(&fe_dpr_test_devp->cdev, &fe_dpr_test_fops);
+  status = cdev_add(&fe_dpr_test_devp->cdev, dev_num, 1);
+  if (status != 0)
+    goto bad_cdev_add;
+ device_obj = device_create_with_groups(cl, NULL, dev_num, NULL, dpr_test_groups, device_name);
+  if (device_obj == NULL)
+    goto bad_device_create;
+  dev_set_drvdata(device_obj, fe_dpr_test_devp);
+  pr_info("dpr_test exit\n");
+  return 0;
+bad_device_create:
+bad_cdev_add:
+  cdev_del(&fe_dpr_test_devp->cdev);
+bad_class_create:
+  class_destroy(cl);
+bad_alloc_chrdev_region:
+  unregister_chrdev_region(dev_num, 1);
+bad_mem_alloc:
+bad_exit_return:
+  pr_info("dpr_test_probe bad exit\n");
+  return ret_val;
+}
+
+
+static int dpr_test_open(struct inode *inode, struct file *file) {
+  // TODO: fill this in (if its needed, it might not be)
+  return 0;
+}
+
+static int dpr_test_release(struct inode *inode, struct file *file) {
+  // TODO: fill this in (if its needed, it might not be)
+  return 0;
+}
+
+static ssize_t dpr_test_read(struct file *file, char *buffer, size_t len, loff_t *offset) {
+  // TODO: fill this in (if its needed, it might not be)
+  return 0;
+}
+
+static ssize_t dpr_test_write(struct file *file, const char *buffer, size_t len, loff_t *offset) {
+  // TODO: fill this in (if its needed, it might not be)
+  return 0;
+}
+static ssize_t name_read(struct device *dev, struct device_attribute *attr, char *buf) {
+  fe_dpr_test_dev_t * devp = (fe_dpr_test_dev_t *)dev_get_drvdata(dev);
+  sprintf(buf, "%s\n", devp->name);
+  return strlen(buf);
+}
+
+static ssize_t dpr_data_read(struct device *dev, struct device_attribute *attr, char *buf) {
+  fe_dpr_test_dev_t * devp = (fe_dpr_test_dev_t *)dev_get_drvdata(dev);
+  fp_to_string(buf, devp->dpr_data, 0, false, 32);
+  strcat2(buf,"\n");
+  return strlen(buf);
+}
+
+static ssize_t dpr_data_write(struct device *dev, struct device_attribute *attr, const char *buf, size_t count) {
+
+  int i,c;
+	uint32_t num;
+  char substring[max_char];
+  int substring_count = 0;
+  int num_counter = 0;
+  
+  fe_dpr_test_dev_t *devp = (fe_dpr_test_dev_t *)dev_get_drvdata(dev);
+
+  // Check to see if the function was called with an empty string
+  if (count < 2)
+    return count;
+    
```

Comment:
I'd like to see this updated at some point to always write the number of values expected by the dual port ram in the FPGA. So a separate property could be made setting the length and then this write function will stop writing when the expected number of inputs is reached and will continue writing the last value until the expected number is reached.